go/interpreter/lfvm/memory_test.go

```diff
@@ -47,3 +52,429 @@ func TestExpansionCosts(t *testing.T) {
 		}
 	}
 }
+
+func TestMemory_CopyWord_CopiesData(t *testing.T) {
+
+	valueSmall := uint256.NewInt(0x1223457890abcdef)
+	valueMiddle := uint256.NewInt(0).Lsh(valueSmall, 64)
+	valueBig := uint256.NewInt(0).Lsh(valueSmall, 256-16)
+	memorySize := uint64(32)
+
+	tests := map[string]struct {
+		offset       uint64
+		expectedData *uint256.Int
+	}{
+		"regular": {
+			offset:       0,
+			expectedData: valueSmall},
+		"small offset": {
+			offset:       memorySize / 4,
+			expectedData: valueMiddle},
+		"big offset crops value": {
+			offset:       memorySize - 2,
+			expectedData: valueBig},
+		"big offset returns zero": {
+			offset:       memorySize,
+			expectedData: uint256.NewInt(0)},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			ctxt := getEmptyContext()
+			m := NewMemory()
+			target := uint256.NewInt(1)
+			m.store = make([]byte, memorySize)
+			copy(m.store[24:], valueSmall.Bytes())
+			ctxt.gas = 100
+
+			err := m.CopyWord(test.offset, target, &ctxt)
+			if err != nil {
+				t.Fatalf("unexpected error, want: %v, got: %v", nil, err)
+			}
+			if target.Cmp(test.expectedData) != 0 {
+				t.Errorf("unexpected target value, want: %x, got: %x", *test.expectedData, *target)
+			}
+		})
+	}
+}
+
+func TestMemory_CopyWord_ReturnsError(t *testing.T) {
+
+	valueSmall := uint256.NewInt(0x1223457890abcdef)
+	baseTargetValue := uint256.NewInt(1)
+	memorySize := uint64(32)
+	enoughGas := tosca.Gas(100)
+
+	tests := map[string]struct {
+		gas           tosca.Gas
+		offset        uint64
+		expectedError error
+	}{
+		"not enough gas": {
+			gas:           6,
+			offset:        uint64(memorySize * 2),
+			expectedError: errOutOfGas},
+		"offset overflow": {
+			gas:           enoughGas,
+			offset:        math.MaxUint64,
+			expectedError: errGasUintOverflow},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			ctxt := getEmptyContext()
+			m := NewMemory()
+			target := baseTargetValue
+			m.store = make([]byte, memorySize)
+			copy(m.store[24:], valueSmall.Bytes())
+			ctxt.gas = test.gas
+
+			err := m.CopyWord(test.offset, target, &ctxt)
+			if !errors.Is(err, test.expectedError) {
+				t.Fatalf("unexpected error, want: %v, got: %v", test.expectedError, err)
+			}
+			if target.Cmp(baseTargetValue) != 0 {
```

Comment:
By default, if there is an error, there should not be any guarantees on the target value unless explicitly documented. So this check may be unnecessary.